core/lib/types/src/transaction_request.rs

```diff
@@ -691,8 +671,21 @@ impl TransactionRequest {
             None => tx.recover_default_signer(default_signed_message).ok(),
         };
 
-        let hash = tx.get_tx_hash_with_signed_message(&default_signed_message, chain_id)?;
+        // `tx.raw` is set, so unwrap is safe here.
+        let hash = tx
+            .get_tx_hash_with_signed_message_raw(&default_signed_message)?
+            .unwrap();
+        Ok((tx, hash))
+    }
 
+    pub fn from_bytes(
+        bytes: &[u8],
+        chain_id: L2ChainId,
+    ) -> Result<(Self, H256), SerializationTransactionError> {
+        let (tx, hash) = Self::from_bytes_unverified(bytes)?;
+        if tx.chain_id.is_some() && tx.chain_id != Some(chain_id.as_u64()) {
```

Comment:
Note, that for some types having chain id is required. It is actually required for all types except for the legacy one

Reply:
made chain_id required for every transaction type except for legacy.